base/applications/osk/main.c

```diff
@@ -178,17 +178,49 @@ int OSK_DlgInitDialog(HWND hDlg)
     monitor = MonitorFromPoint(Pt, MONITOR_DEFAULTTOPRIMARY);
     info.cbSize = sizeof(info);
     GetMonitorInfoW(monitor, &info);
-
-    /* Move the dialog on the bottom of main screen */
     GetWindowRect(hDlg, &rcWindow);
-    MoveWindow(hDlg,
-               (info.rcMonitor.left + info.rcMonitor.right) / 2 - // Center of screen
-                   (rcWindow.right - rcWindow.left) / 2,          // - half size of dialog
-               info.rcMonitor.bottom -               // Bottom of screen
-                   (rcWindow.bottom - rcWindow.top), // - size of window
-               rcWindow.right - rcWindow.left,     // Width
-               rcWindow.bottom - rcWindow.top,     // Height
-               TRUE);
+
+    /* 
+        Get the screen info based on the dialog handle.
+        This will be used to check if the dialog is outside the working area of the monitor.
+    */
+    monitorDlg = MonitorFromWindow(hDlg, MONITOR_DEFAULTTONEAREST);
```

Comment:
You may be able to re-use the `monitor` variable here instead of introducing a new one, since `monitor` is not used at all in the code that follows below.